plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java

```diff
@@ -296,6 +307,10 @@ public void updateClasspath(String projectUri, List<ClasspathEntry> entries) {
    * @return source folders
    */
   public List<String> getSourceFolders(String projectPath) {
+    if (!findInitializedLanguageServer().isPresent()) {
```

Comment:
Maybe it is worth to move this check to separate method?

Reply:
Why?

Reply:
because we have at least 2 exact checks

Reply:
IMO, the code would be a bit clearer

Reply:
Thx for advice, but I disagree.